hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/SnapshotFileCache.java

```diff
@@ -184,7 +184,7 @@ public synchronized void triggerCacheRefreshForTesting() {
   // XXX this is inefficient to synchronize on the method, when what we really need to guard against
   // is an illegal access to the cache. Really we could do a mutex-guarded pointer swap on the
   // cache, but that seems overkill at the moment and isn't necessarily a bottleneck.
-  public synchronized Iterable<FileStatus> getUnreferencedFiles(Iterable<FileStatus> files,
+  public Iterable<FileStatus> getUnreferencedFiles(Iterable<FileStatus> files,
```

Comment:
I think we should store cache into a local variable and use it in the method, as it may be switched during the method execution.

Reply:
Thanks, I'll change it.

Reply:
Yes, it's a good idea.